imports/wasi_snapshot_preview1/testdata/zig-cc/wasi.c

```diff
@@ -171,6 +171,9 @@ void main_nonblock(char* fpath) {
   ssize_t newLen = 0;
   while (newLen == 0) {
     newLen = read(fd, buf, sizeof(buf));
+    if strlen(buf) == 0 {
+      newLen = 0;
+    }
```

Comment:
comment on the rationale?

Reply:
oh right, I added a comment inline. If buf contains an empty string (like in the case of #1994) the length of the buffer might be 1 (because of the terminator `\0`) but we want to discard the element and stay in the loop.

Reply:
(and apparently I also forgot to push the wasm)